setup.py

```diff
@@ -12,8 +12,24 @@
 from setuptools import setup
 from setuptools.extension import Extension
 
-include_dirs = [os.path.dirname(get_python_inc())]
 library_dirs = [get_config_var("LIBDIR")]
+
+# if the 'libucx' wheel was installed, add its library location to
+# library_dirs so it'll be linked against
+#
+# NOTE: doing this '.__file__' stuff because `sysconfig.get_path("platlib")` and similar
+#       can return paths to the main Python interpreter's installation locations...
+#       not where 'libucx' is going to be installed at build time when using
+#       build isolation
+try:
+    import libucx
+
+    libucx_libdir = os.path.join(os.path.dirname(libucx.__file__), "lib")
+    library_dirs.append(libucx_libdir)
```

Comment:
Without this, builds in an environment without a system installation of `libuc{m,p,s,t}` fail like this:

```text
[1/1] Cythonizing ucp/_libs/ucx_api.pyx
building 'ucp._libs.ucx_api' extension
creating build/temp.linux-x86_64-cpython-310
creating build/temp.linux-x86_64-cpython-310/ucp
creating build/temp.linux-x86_64-cpython-310/ucp/_libs
creating build/temp.linux-x86_64-cpython-310/ucp/_libs/src
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/pyenv/versions/3.10.14/include -I/pyenv/versions/3.10.14/include/python3.10 -c ucp/_libs/src/c_util.c -o build/temp.linux-x86_64-cpython-310/ucp/_libs/src/c_util.o -std=c99 -Werror
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/pyenv/versions/3.10.14/include -I/pyenv/versions/3.10.14/include/python3.10 -c ucp/_libs/ucx_api.c -o build/temp.linux-x86_64-cpython-310/ucp/_libs/ucx_api.o -std=c99 -Werror
gcc -pthread -shared -L/pyenv/versions/3.10.14/lib -Wl,-rpath,/pyenv/versions/3.10.14/lib -L/usr/lib -L/pyenv/versions/3.10.14/lib -Wl,-rpath,/pyenv/versions/3.10.14/lib -L/usr/lib build/temp.linux-x86_64-cpython-310/ucp/_libs/src/c_util.o build/temp.linux-x86_64-cpython-310/ucp/_libs/ucx_api.o -L/pyenv/versions/3.10.14/lib -L/pyenv/versions/3.10.14/lib -lucp -luct -lucm -lucs -o build/lib.linux-x86_64-cpython-310/ucp/_libs/ucx_api.cpython-310-x86_64-linux-gnu.so
/opt/rh/gcc-toolset-11/root/usr/bin/ld: cannot find -lucp
/opt/rh/gcc-toolset-11/root/usr/bin/ld: cannot find -luct
/opt/rh/gcc-toolset-11/root/usr/bin/ld: cannot find -lucm
/opt/rh/gcc-toolset-11/root/usr/bin/ld: cannot find -lucs
collect2: error: ld returned 1 exit status
error: command '/opt/rh/gcc-toolset-11/root/usr/bin/gcc' failed with exit code 1
```

Reply:
Ah yeah I completely forgot that ucx-py is not built with scikit-build-core or CMake, so we need to provide the library location another way. Rather than rewriting this logic manually, which makes ucx-py sensitive to changes in the libucx wheel, I think you could instead do something like what I've proposed as a change.